pallets/multisig/src/benchmarking.rs

```diff
@@ -0,0 +1,284 @@
+        generate_signers::<T>(&mut signers, i as usize);
+    }: _(caller.origin, signers, i as u64)
+
+    create_or_approve_proposal_as_identity {
+        // Number of signers
+        let i in 1 .. T::MaxSigners::get();
+
+        let alice = <UserBuilder<T>>::default().build_with_did("alice", 0);
+        let mut signers = vec![Signatory::from(alice.did())];
+        generate_signers::<T>(&mut signers, i as usize);
+        let multisig = generate_multisig::<T>(alice.account(), alice.origin(), signers)?;
+    }: _(alice.origin, multisig, Box::new(frame_system::Call::<T>::remark(vec![]).into()), None, true)
+
+    create_or_approve_proposal_as_key {
+        // Number of signers
+        let i in 1 .. T::MaxSigners::get();
+
+        let alice = <UserBuilder<T>>::default().build_with_did("alice", 0);
+        let bob = <UserBuilder<T>>::default().build("bob", 0);
+        let mut signers = vec![Signatory::Account(bob.account())];
+        generate_signers::<T>(&mut signers, i as usize);
+        let multisig = generate_multisig::<T>(alice.account(), alice.origin(), signers)?;
+    }: _(bob.origin, multisig, Box::new(frame_system::Call::<T>::remark(vec![]).into()), None, true)
+
+    create_proposal_as_identity {
+        // Number of signers
+        let i in 1 .. T::MaxSigners::get();
+
+        let alice = <UserBuilder<T>>::default().build_with_did("alice", 0);
+        let mut signers = vec![Signatory::from(alice.did())];
+        generate_signers::<T>(&mut signers, i as usize);
+        let multisig = generate_multisig::<T>(alice.account(), alice.origin(), signers)?;
+    }: _(alice.origin, multisig, Box::new(frame_system::Call::<T>::remark(vec![]).into()), None, true)
+
+    create_proposal_as_key {
+        // Number of signers
+        let i in 1 .. T::MaxSigners::get();
+
+        let alice = <UserBuilder<T>>::default().build_with_did("alice", 0);
+        let bob = <UserBuilder<T>>::default().build("bob", 0);
+        let mut signers = vec![Signatory::Account(bob.account())];
+        generate_signers::<T>(&mut signers, i as usize);
+        let multisig = generate_multisig::<T>(alice.account(), alice.origin(), signers)?;
+    }: _(bob.origin, multisig, Box::new(frame_system::Call::<T>::remark(vec![]).into()), None, true)
+
+    approve_as_identity {
+        let alice = <UserBuilder<T>>::default().build_with_did("alice", 0);
+        let bob = <UserBuilder<T>>::default().build_with_did("bob", 0);
+        let signers = vec![
+            Signatory::from(alice.did()),
+            Signatory::from(bob.did()),
+        ];
+        let multisig = generate_multisig::<T>(alice.account(), alice.origin(), signers)?;
+        <MultiSig<T>>::create_proposal_as_identity(
+            alice.origin(),
+            multisig.clone(),
+            Box::new(frame_system::Call::<T>::remark(vec![]).into()),
+            None,
+            true
+        )?;
+    }: _(bob.origin, multisig, 0)
```

Comment:
This currently executes the proposal as well. We need to eventually split the execution from approval. I'll capture it in Jira.